tonconnect-server/TonConnect2.md

```diff
@@ -0,0 +1,533 @@
+# TON Connect 2
+
+* [Overview](#Overview)
+* [Workflows](#Workflows)
+* [Compatibility](#Compatibility)
+* [Definitions](#Definitions)
+* [Requests and Responses](#Requests-and-Responses)
+
+## Overview
+
+TON blockchain enables creation of trust-minimized applications and services.
+
+TON apps may control various assets (coins, tokens, NFTs etc.) according to the publicly auditable and immutable logic performed by the blockchain. This way users do not have to place trust in closed private systems to not censor or steal their money.
+
+Access to TON apps is controlled by the wallets: regular applications that keep users’ cryptographic keys on their devices. Wallets enable users to sign into applications and authorize blockchain transactions.
+
+The goal of Ton Connect is to enable smooth and safe interaction between wallets, services and apps.
+
+
+### Apps vs services
+
+In this specification we use the following terminology:
+
+**App** is a software that runs on behalf of the user and does not rely on a dedicated backend server for accessing users’ data.
+
+**Service** is a software that runs on behalf of its operator, stores users’ data and regulates access to the data.
```

Comment:
I intentionally went on with this radical approach to see the reaction from reviewers. 

Custodial/non-custodial is a orthogonal to this issue because even centralized services may be non-custodial. The distinction that matters to TonConnect and the wallets is between (a) the software runs on behalf of the user and does not have its own secure identity, and (b) has its own identity and runs on someone else's machine. In case of (a) we don't have to prove anything in either direction, but have to employ identity indirectly through smart contracts. In case of (b) we have well-defined 3rd party to send responses to.

We may clarify the terms as "standalone app" and a "server", maybe?

Reply:
I actually like the wording a lot. I think @oleganza did a great job here. The distinction between the two use-cases is indeed radical. People who read the standard need a very very good reason to write a *service*. The majority of them should write an *app*. Most teams in the TON ecosystem are going to be anonymous and not very trustworthy, so we should not encourage them to hold any positions that require trust.